src/DSE.Open/MemoryExtensions.cs

```diff
@@ -253,4 +257,26 @@ public static int Replace<T>(this Span<T> span, Func<T, bool> predicate, T repla
 
         return count;
     }
+
+    /// <summary>
+    /// Computes the sum of the sequence of Int32 values that are obtained by invoking a transform function
+    /// on each element of the input sequence.
+    /// </summary>
+    public static TResult Sum<TSource, TResult>(this ReadOnlySpan<TSource> source, Func<TSource, TResult> selector)
```

Comment:
Gather INumber-related extensions in NumberExtensions - rather than MemoryExtensions?

Or make MemoryExtensions partial and split to a MemoryExtensions.Number.cs file.

I can imagine Max, Min ([and many more](https://github.com/mathnet/mathnet-numerics/blob/master/src/Numerics/Statistics/ArrayStatistics.cs)) over ReadOnlySpan<T> where T : INumber

Reply:
I'm not quite sure on the distinction. `Min` and `Max` make more sense, because they're constraining the input span to `INumber` or similar. `Sum` doesn't constrain the source in the same way, it's just the result that is a generic number.